Node/src/TeamsChatConnector.ts

```diff
@@ -182,27 +189,41 @@ export class TeamsChatConnector extends builder.ChatConnector {
     for (var event of events) {
       let invoke = <IInvokeEvent>event;
       if (invoke.type == 'invoke') {
-        let query = <ComposeExtensionQuery>(<any>event).value;
-        let handler: ComposeExtensionHandlerType;
+        let compExthandler: ComposeExtensionHandlerType;
+        let o365Handler: O365ConnectorCardActionHandlerType;
         switch (invoke.name) {
           case TeamsChatConnector.queryInvokeName:
-            handler = this.dispatchQuery.bind(this);
+            compExthandler = this.dispatchQuery.bind(this);
             break;
           case TeamsChatConnector.querySettingUrlInvokeName:
-            handler = this.querySettingsUrlHandler.bind(this);
+            compExthandler = this.querySettingsUrlHandler.bind(this);
             break;
           case TeamsChatConnector.settingInvokeName:
           {
-            handler = this.settingsUpdateHandler.bind(this);
+            compExthandler = this.settingsUpdateHandler.bind(this);
             break;
           }
+          case TeamsChatConnector.o365CardActionInvokeName:
+            o365Handler = this.o365CardActionHandler.bind(this);
+            break;
           default:
             realEvents.push(event);
             break;
         }
-        if (handler) {
+        if (compExthandler) {
```

Comment:
Let's add some (judicious) blank lines between the blocks to this method, it's getting rather hard to read as a wall of text.